include/seqan3/alignment/pairwise/align_result_selector.hpp

```diff
@@ -116,8 +119,39 @@ struct align_result_selector
         }
     }
 
-    //!\brief The determined result type.
-    using type = decltype(_determine());
+    //!\brief Augments the result type with the debug output for score and trace matrix.
+    template <typename alignment_result_value_t>
+    static constexpr auto augment_if_debug(alignment_result_value_t)
+    {
+        if constexpr (!configuration_t::template exists<align_cfg::debug>())
+        {
+            return alignment_result_value_t{};
+        }
+        else
+        {
+            using as_type_list = transfer_template_args_onto_t<alignment_result_value_t, type_list>;
+            using score_matrix_t = two_dimensional_matrix<int32_t, std::allocator<int32_t>, matrix_major_order::column>;
+            using trace_matrix_t = two_dimensional_matrix<trace_directions,
+                                                        std::allocator<trace_directions>,
+                                                        matrix_major_order::column>;
+
+            if constexpr (configuration_t::template exists<align_cfg::result<with_alignment_type>>())
+            {
+                using with_score_t = list_traits::replace_at<score_matrix_t, 5, as_type_list>;
+                return transfer_template_args_onto_t<list_traits::replace_at<trace_matrix_t, 6, with_score_t>,
+                                                     alignment_result_value_type>{};
+            }
+            else
+            {
+                return transfer_template_args_onto_t<list_traits::replace_at<score_matrix_t, 5, as_type_list>,
```

Comment:
When I set the debug config, I receive always the score matrix, and only if the alignment was requested I additionally receive the trace matrix. Please describe this behaviour in the documentation.

Reply:
It is?
> Using this configuration allows to output the alignment score matrix and if requested the alignment trace matrix.\

I rephrase it to make it more clear.